src/index.ts

```diff
@@ -0,0 +1,775 @@
+/**
+ * A Result model, loosely based on the Rust result type.
+ *
+ * @module
+ */
+
+/**
+ * `Result<T, E>` is used for returning and propagating errors.
+ *
+ * It can be constructed with {@link Ok} or {@link Err}.
+ *
+ * @typeParam T - The type of the `Ok` value.
+ * @typeParam E - The type of the `Err` value.
+ */
+export class Result<T, E> {
+  /** @internal */
+  // biome-ignore lint/complexity/noUselessConstructor: hiding from documentation
+  constructor() {}
+
+  /**
+   * Returns `true` if the result is `Ok`.
+   *
+   * @example
+   *
+   * ```typescript
+   * const x: Result<number, string> = Ok(2)
+   * x.isOk() === true
+   *
+   * const y: Result<number, string> = Err('hey')
+   * y.isOk() === false
+   * ```
+   */
+  isOk(): boolean {
+    throw new Error('Abstract Result')
```

Comment:
I’m assuming that all the `throw new Error('Abstract Result')` are normal for the generic code, and indicate someone forgot to implement something in a concrete thing.

Reply:
Yes and no. I *could* fix this by declaring it as `abstract class Result<T, E>`, but that would change the documentation. I will be experimenting with some other options when I have time (not now) where I do it as a Typescript `interface` that is implemented by `OkResult` and `ErrResult`, neither of which should be documented, but throwing the exceptions makes it really easy to say 'don't instantiate this'.